src/transformers/training_args.py

```diff
@@ -1097,13 +1098,22 @@ def __post_init__(self):
         if self.deepspeed:
             # - must be run very last in arg parsing, since it will use a lot of these settings.
             # - must be run before the model is created.
+            if not is_accelerate_available():
+                raise ValueError("--deepspeed requires Accelerate to be installed: `pip install accelerate`.")
             from transformers.deepspeed import HfTrainerDeepSpeedConfig
 
             # will be used later by the Trainer
             # note: leave self.deepspeed unmodified in case a user relies on it not to be modified)
             self.hf_deepspeed_config = HfTrainerDeepSpeedConfig(self.deepspeed)
             self.hf_deepspeed_config.trainer_config_process(self)
 
+            from accelerate import Accelerator
+            from accelerate.utils import DeepSpeedPlugin
+
+            _ = Accelerator(
+                deepspeed_plugin=DeepSpeedPlugin(hf_ds_config=self.hf_deepspeed_config, zero3_init_flag=True)
+            )
```

Comment:
Same here, we shouldn't define an `Accelerator` here. Since all `AcceleratorState` share the same inner dict, setting its deepspeed plugin directly will work normally.

Reply:
In `training_args.py` when user is running trainer scripts, Accelerator object is nowhere created and hence if we only have to create `AcceleratorState` then we have to pass `_from_accelerator=True` to it. Is that fine?

Reply:
It's not ideal, as that creation should be done by the user only at the right point in their scripts. However it's unlikely a user using the Trainer will re-use the Accelerator object so it can work if there is really no alternative.